From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Wed, 5 Jan 2022 16:43:52 +1000
Subject: [PATCH] Send server port to the master server


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index bd57b74af195c38641406efc933782ed203a323f..4e58ba60d6c86ef490f947fb8c4552c2ff3dc19b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -276,6 +276,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
             this.setPort(dedicatedserverproperties.serverPort);
         }
         bindAddress = new java.net.InetSocketAddress(inetaddress, this.getPort());
+        MultiPaper.setPort(this.getPort()); // MultiPaper
         }
         // Paper end - Unix domain socket support
 
diff --git a/src/main/java/puregero/multipaper/MultiPaper.java b/src/main/java/puregero/multipaper/MultiPaper.java
index fd95c272ca591d034ac6fe6de4a90ce2c2149627..0bfd56aaf6504a87b6a0c60770f12e87adb9a912 100644
--- a/src/main/java/puregero/multipaper/MultiPaper.java
+++ b/src/main/java/puregero/multipaper/MultiPaper.java
@@ -603,4 +603,8 @@ public class MultiPaper {
     public static void chunkChangedStatus(ServerLevel level, ChunkPos pos, ChunkStatus status) {
         getConnection().send(new ChunkChangedStatusMessage(level.getWorld().getName(), pos.x, pos.z, BuiltInRegistries.CHUNK_STATUS.getKey(status).toString()));
     }
+
+    public static void setPort(int port) {
+        getConnection().setPort(port);
+    }
 }
diff --git a/src/main/java/puregero/multipaper/MultiPaperConnection.java b/src/main/java/puregero/multipaper/MultiPaperConnection.java
index 39dbf785a45b51d0a56c21699cace7bde9be6593..043b23e15725cd588a809a2c0b4d066243256c58 100644
--- a/src/main/java/puregero/multipaper/MultiPaperConnection.java
+++ b/src/main/java/puregero/multipaper/MultiPaperConnection.java
@@ -64,6 +64,7 @@ public class MultiPaperConnection extends ServerBoundMessageHandler {
     private final Map<String, ExternalServer> serversMap = new ConcurrentHashMap<>();
     public String secret;
     public Map<String, byte[]> dataCache = Maps.newHashMap();
+    public int port = -1;
 
     public MultiPaperConnection() {
         System.setProperty("multipaper.netty.threads", System.getProperty("multipaper.netty.threads", Integer.toString(Math.min(Runtime.getRuntime().availableProcessors(), 3))));
@@ -102,6 +103,10 @@ public class MultiPaperConnection extends ServerBoundMessageHandler {
         LOGGER.info("Connected to " + channel.remoteAddress());
         channel.write(new HelloMessage(myName, thisServersUuid));
 
+        if (port >= 0) {
+            channel.write(new SetPortMessage(port));
+        }
+
         for (MasterBoundMessage unhandledRequest : unhandledRequests) {
             channel.write(unhandledRequest);
         }
@@ -193,6 +198,14 @@ public class MultiPaperConnection extends ServerBoundMessageHandler {
         return serversMap;
     }
 
+    public void setPort(int port) {
+        if (!MultiPaperConfiguration.get().masterConnection.advertiseToBuiltInProxy) {
+            return;
+        }
+        this.port = port;
+        send(new SetPortMessage(port));
+    }
+
     @Override
     public void handle(ServerInfoUpdateMessage message) {
         ExternalServer server = serversMap.computeIfAbsent(message.name, key -> new ExternalServer(key, key.equals(myName)));
