From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Tue, 8 Mar 2022 00:37:35 +1000
Subject: [PATCH] Only drop leads on the local server


diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
index 6bea7e9aac6d7269dbee525a197cfe826387f3de..8b77cdb9fed0ac7d7055de6b2bcd4e43289c547a 100644
--- a/src/main/java/net/minecraft/world/entity/Mob.java
+++ b/src/main/java/net/minecraft/world/entity/Mob.java
@@ -99,6 +99,7 @@ import org.bukkit.event.entity.EntityTargetEvent;
 import org.bukkit.event.entity.EntityTransformEvent;
 import org.bukkit.event.entity.EntityUnleashEvent;
 import org.bukkit.event.entity.EntityUnleashEvent.UnleashReason;
+import puregero.multipaper.MultiPaper;
 // CraftBukkit end
 
 public abstract class Mob extends LivingEntity implements EquipmentUser, Targeting {
@@ -1679,7 +1680,7 @@ public abstract class Mob extends LivingEntity implements EquipmentUser, Targeti
             this.leashHolder = null;
             this.delayedLeashInfo = null;
             this.clearRestriction();
-            if (!this.level().isClientSide && dropItem) {
+            if (!this.level().isClientSide && dropItem && MultiPaper.isChunkLocal(this.level().getChunkIfLoaded(blockPosition()))) { // MultiPaper - only drop leads on the local server
                 this.forceDrops = true; // CraftBukkit
                 this.spawnAtLocation((ItemLike) Items.LEAD);
                 this.forceDrops = false; // CraftBukkit
