From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Wed, 6 Apr 2022 20:30:56 +1000
Subject: [PATCH] A view distance of 1 also works


diff --git a/src/main/java/io/papermc/paper/chunk/system/RegionizedPlayerChunkLoader.java b/src/main/java/io/papermc/paper/chunk/system/RegionizedPlayerChunkLoader.java
index fede16eddfe1ce55af81d68a6a1d36c79c5a6462..a8f62b3282fd5994bf5e81b716af81943c537c32 100644
--- a/src/main/java/io/papermc/paper/chunk/system/RegionizedPlayerChunkLoader.java
+++ b/src/main/java/io/papermc/paper/chunk/system/RegionizedPlayerChunkLoader.java
@@ -122,7 +122,7 @@ public class RegionizedPlayerChunkLoader {
 
     public static final TicketType<Long> REGION_PLAYER_TICKET = TicketType.create("region_player_ticket", Long::compareTo);
 
-    public static final int MIN_VIEW_DISTANCE = 2;
+    public static final int MIN_VIEW_DISTANCE = 1; // MultiPaper - minimum can actually be 1
     public static final int MAX_VIEW_DISTANCE = 32;
 
     public static final int TICK_TICKET_LEVEL = 31;
