From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Wed, 13 Apr 2022 16:37:10 +1000
Subject: [PATCH] Only run PlayerArmorChangeEvent for local players


diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index 49350ecf86eb9a2d270694dadc5e0d1f490188f1..832efae8d8989fb1e804c13e8f1f8086baada715 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -3318,7 +3318,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
 
             if (this.equipmentHasChanged(itemstack1, itemstack2)) {
                 // Paper start - PlayerArmorChangeEvent
-                if (this instanceof ServerPlayer && enumitemslot.getType() == EquipmentSlot.Type.ARMOR) {
+                if (MultiPaper.isRealPlayer(this) && enumitemslot.getType() == EquipmentSlot.Type.ARMOR) { // MultiPaper - only run PlayerArmorChangeEvent for local players
                     final org.bukkit.inventory.ItemStack oldItem = CraftItemStack.asBukkitCopy(itemstack1);
                     final org.bukkit.inventory.ItemStack newItem = CraftItemStack.asBukkitCopy(itemstack2);
                     new com.destroystokyo.paper.event.player.PlayerArmorChangeEvent((Player) this.getBukkitEntity(), com.destroystokyo.paper.event.player.PlayerArmorChangeEvent.SlotType.valueOf(enumitemslot.name()), oldItem, newItem).callEvent();
