From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Tue, 5 Jul 2022 17:44:09 +1000
Subject: [PATCH] Tick sleepCounter for external players


diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 03da428dc7f8a479402376b4d0243f43a1490581..456e8a934c3b26e51952e3b1b09854ffdaab2ed0 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -173,7 +173,7 @@ public abstract class Player extends LivingEntity {
     public double xCloak;
     public double yCloak;
     public double zCloak;
-    public int sleepCounter;
+    public int sleepCounter; public int getSleepCounter() { return this.sleepCounter; } public void setSleepCounter(int sleepCounter) { this.sleepCounter = sleepCounter; } // MultiPaper - add getter and setter
     protected boolean wasUnderwater;
     private final Abilities abilities = new Abilities();
     public int experienceLevel;
diff --git a/src/main/java/puregero/multipaper/MultiPaper.java b/src/main/java/puregero/multipaper/MultiPaper.java
index 5ba3199f7977568779df0cc51c735af0cbbb28c4..91f2ee20d9001dcf4fb98c90f559a76604103226 100644
--- a/src/main/java/puregero/multipaper/MultiPaper.java
+++ b/src/main/java/puregero/multipaper/MultiPaper.java
@@ -97,6 +97,12 @@ public class MultiPaper {
                 }
 
                 externalPlayer.tickDeathIfDead();
+
+                if (externalPlayer.isSleeping()) {
+                    externalPlayer.setSleepCounter(Math.min(100, externalPlayer.getSleepCounter() + 1));
+                } else {
+                    externalPlayer.setSleepCounter(0);
+                }
             }
 
             player.syncExperience();
