From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Tue, 5 Jul 2022 17:48:07 +1000
Subject: [PATCH] Only wake up local players


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index e50fb73f011bb2855c2be6bdd7ed86293e6752c3..c6178b341c897e6b82d15fbe23b2d04e4e2932e9 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1034,7 +1034,7 @@ this.blockTicks = new LevelTicks<>(this, this::isPositionTickingWithEntitiesLoad
 
     private void wakeUpAllPlayers() {
         this.sleepStatus.removeAllSleepers();
-        (this.players.stream().filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((entityplayer) -> { // CraftBukkit - decompile error
+        (this.players.stream().filter(MultiPaper::isRealPlayer).filter(LivingEntity::isSleeping).collect(Collectors.toList())).forEach((entityplayer) -> { // CraftBukkit - decompile error // MultiPaper - only wake up local players
             entityplayer.stopSleepInBed(false, false);
         });
     }
