--- a/purpur-server/build.gradle.kts
+++ b/purpur-server/build.gradle.kts
@@ -33,7 +33,23 @@ paperweight {
             outputDir = rootDirectory.dir("paper-server")
         }
     }
-    activeFork = purpur
+    // MultiPaper start
+    val multipaper = forks.register("multipaper") {
+        forks = purpur
+        upstream.patchRepo("paperServer") {
+            upstreamRepo = purpur.patchedRepo("paperServer")
+            patchesDir = rootDirectory.dir("patches/${providers.gradleProperty("purpurVersion").getOrElse("26.3")}/server")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+        upstream.patchDir("purpurServer") {
+            upstreamPath = "purpur-server"
+            excludes = setOf("src/minecraft", "paper-patches", "minecraft-patches", "build.gradle.kts", "build.gradle.kts.patch")
+            patchesDir = rootDirectory.dir("patches/${providers.gradleProperty("purpurVersion").getOrElse("26.3")}/server/purpur")
+            outputDir = rootDirectory.dir("purpur-server")
+        }
+    }
+    activeFork = multipaper
+    // MultiPaper end
     // Purpur end - Rebrand
 
     /*updatingMinecraft { // Purpur start - Rebrand - we don't need this
@@ -105,10 +121,14 @@ sourceSets {
     main {
         java { srcDir("../paper-server/src/main/java") }
         resources { srcDir("../paper-server/src/main/resources") }
+        java { srcDir("../purpur-server/src/main/java") } // MultiPaper
+        resources { srcDir("../purpur-server/src/main/resources") } // MultiPaper
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
+        java { srcDir("../purpur-server/src/test/java") } // MultiPaper
+        resources { srcDir("../purpur-server/src/test/resources") } // MultiPaper
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -137,7 +157,10 @@ abstract class MockitoAgentProvider : CommandLineArgumentProvider {
 }
 
 dependencies {
-    implementation(project(":purpur-api")) // Purpur
+    implementation(project(":multipaper-mastermessagingprotocol")) // MultiPaper
+    implementation(project(":multipaper-api")) // Purpur // MultiPaper
+    implementation("commons-lang:commons-lang:2.6") // MultiPaper
+    implementation("io.methvin:directory-watcher:0.15.0") // MultiPaper
     implementation("ca.spottedleaf:leafpile:1.2.0")
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.27.1") // fall back to jni on java 21
